airflow/www/views.py

```diff
@@ -1687,8 +1687,9 @@ def refresh(self, session=None):
     @action_logging
     def refresh_all(self):
         dagbag.collect_dags(only_if_updated=False)
-        # sync permissions for all dags
-        appbuilder.sm.sync_perm_for_dag()
+        for dag_id in dagbag.dags:
+            # sync permissions for all dags
+            appbuilder.sm.sync_perm_for_dag(dag_id)
```

Comment:
@astahlman , since you are at this function, maybe we should consider add a check and see whether the user has permission to do refresh_all view.

The AirflowSecurityManager provides a way to check whether the user's role(https://github.com/apache/airflow/blob/master/airflow/www/security.py#L283). But we could do it in a different pr.